docs/configuration.asciidoc

```diff
@@ -15,20 +13,319 @@ server_url: 'http://localhost:8200'
+
+If the current envrironment isn't included, the agent will effectively be a _noop_
+and do nothing.
+So you can keep in your custom instrumentation code without the agent sending
+anything nor complain.
+
+[float]
+[[config-framework-name]]
+==== `framework_name`
+[options="header"]
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_FRAMEWORK_NAME` | `framework_name` | Depending on framework
+|============
+
+Name of the used framework.
+For Rails or Sinatra, this defaults to `Ruby on Rails` and `Sinatra` respectively,
+otherwise defaults to `nil`.
+
+[float]
+[[config-framework-version]]
+==== `framework_version`
+[options="header"]
+|============
+| Environment                     | `Config` key        | Default
+| `ELASTIC_APM_FRAMEWORK_VERSION` | `framework_version` | Depending on framework
+|============
+
+Version number of the used framework.
+For Ruby on Rails and Sinatra, this defaults to the used version of the framework,
+otherwise, the default is `nil`.
+
+[float]
+[[config-hostname]]
+==== `hostname`
+
+[options="header"]
+|============
+| Environment                | `Config` key  | Default    | Example
+| `ELASTIC_APM_HOSTNAME`     | `hostname`    | `hostname` | `app-server01.example.com`
+|============
+
+The host name to use when sending error and transaction data to the APM server.
+
+[float]
+[[config-source-lines-error-app-frames]]
+==== `source_lines_error_app_frames`
+[float]
+[[config-source-lines-error-library-frames]]
+==== `source_lines_error_library_frames`
+[float]
+[[config-source-lines-span-app-frames]]
+==== `source_lines_span_app_frames`
+[float]
+[[config-source-lines-span-library-frames]]
+==== `source_lines_span_library_frames`
+
+|============
+| Environment                                     | `Config` key                        | Default
+| `ELASTIC_APM_SOURCE_LINES_ERROR_APP_FRAMES`     | `source_lines_error_app_frames`     | `5`
+| `ELASTIC_APM_SOURCE_LINES_ERROR_LIBRARY_FRAMES` | `source_lines_error_library_frames` | `5`
+| `ELASTIC_APM_SOURCE_LINES_SPAN_APP_FRAMES`      | `source_lines_span_app_frames`      | `0`
+| `ELASTIC_APM_SOURCE_LINES_SPAN_LIBRARY_FRAMES`  | `source_lines_span_library_frames`  | `0`
+|============
+
+By default, the APM agent collects source code snippets for errors.
+With the above settings, you can modify how many lines of source code is collected.
+
+We differ between errors and spans, as well as library frames and app frames.
+
+WARNING: Especially for spans, collecting source code can have a large impact on
+storage use in your Elasticsearch cluster.
+
+[float]
+[[config-max-queue-size]]
+==== `max_queue_size`
+
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_MAX_QUEUE_SIZE` | `max_queue_size` | `500`
+|============
+
+Maximum queue length of transactions before sending transactions to the APM server.
+A lower value will increase the load on your APM server,
```

Comment:
I would not state that, as it would lead to smaller request bodies and a more consistent load on the Server, so I'd actually expect this to have a good impact on the server.
It might increase the load on the agent for sending more http requests than otherwise.